scriptlets/git_get

```diff
@@ -0,0 +1,82 @@
+#!/usr/bin/env bash
+
+# Perform a light, shallow fetch of a single commit or branch from a repo
+
+# disable tracing (if previously enabled)
+[[ "$-" == *x* ]] && TRACING=true && set +x || TRACING=false
+
+usage() {
+    echo "Usage: $(basename ${BASH_SOURCE[0]}) <repository> [commit [<commit_ref>]]"
+    echo "Usage: $(basename ${BASH_SOURCE[0]}) <repository> branch <branch>"
+}
+
+if [[ "$#" -lt 1 ]]; then
+    echo "Error: no repository specified"
+    usage
+    exit 1
+fi
+
+REPO="$1"
+shift
+
+COMMIT_DEFAULT="HEAD"
+COMMIT=""
+BRANCH=""
+while [[ "$#" -gt 0 ]]; do
+    case $1 in
+        commit)
+            if [ -z "$COMMIT" ]; then
+                if [ -n "$2" ]; then
+                    COMMIT=$2
+                    shift
+                else
+                    COMMIT="$COMMIT_DEFAULT"
+                fi
+            else
+                echo "Error: commit already specified"
+                usage
+                exit 1
+            fi
+            ;;
+        branch)
+            if [ -z "$BRANCH" ]; then
+                if [ -n "$2" ]; then
+                    BRANCH=$2
+                    shift
+                else
+                    echo "Error: value required for branch"
+                    usage
+                    exit 1
+                fi
+            else
+                echo "Error: branch already specified"
+                usage
+                exit 1
+            fi
+            ;;
+        *)
+            echo "Error: Invalid argument $1";
+            usage
+            exit 1
+            ;;
+    esac
+    shift
+done
```

Comment:
imo this is flying to close to the sun as far as maximum complexity I would like to see in a bash script.  Still, if you want to keep it in bash consider using `getopts` and please fix the fact that I can provide both a commit and a branch to this it will not complain!

Reply:
That's a fair point regarding length but most of it is argument parsing, so hopefully not too complex. Using `getopts` is a great suggestion but in this particular case I don't think it would really help as there are no short-option arguments. Your comment did get me thinking though and I have slightly refactored to hopefully make the argument parsing clearer (i.e. to better reflect the existence of two different "modes".